File: 0080-remove-duplicates-from-sorted-array-ii/0080-remove-duplicates-from-sorted-array-ii.java
class Solution {
    public int removeDuplicates(int[] nums) {
        if(nums.length <= 2)
            return 2;
        int i = 1;
        int count = 1;

        for(int j = 1; j < nums.length; j++) {
            if(nums[j - 1] == nums[j]) {
                count++;
            } else {
                count = 1;
            }

            if(count <= 2) {
                nums[i++] = nums[j];
            }
        }

        return i;
    }
}

File: 0080-remove-duplicates-from-sorted-array-ii/0080-remove-duplicates-from-sorted-array-ii.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        
        if(nums.size() <= 2) {
          return nums.size();
        }
        int slow = 2; int fast = 2;

        while(fast < nums.size()) {
          if(nums[slow - 2] == nums[fast]) {
            fast++;
          } else {
            nums[slow++] = nums[fast++];
          }
        }

        return slow;
    }
};